components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dao/impl/IdentityUserMetadataMgtDAOImpl.java

```diff
@@ -0,0 +1,508 @@
+                if (cachedUserIdentityDTO != null) {
+                    cachedUserIdentityDTO.getUserIdentityDataMap().putAll(userIdentityDTO.getUserIdentityDataMap());
+                    identityDataStoreCache.addToCache(key, cachedUserIdentityDTO, tenantId);
+                } else {
+                    identityDataStoreCache.addToCache(key, userIdentityDTO, tenantId);
+                }
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            log.error("Error while obtaining tenant ID from user store manager", e);
+        }
+    }
+
+    /**
+     * Store user metadata in the user store.
+     *
+     * @param userStoreManager          User store manager
+     * @param username                  Username
+     * @param tenantId                  Tenant ID
+     * @param claimURI                  Claim URI
+     * @param value                     Claim attribute value
+     * @param eventName                 Event name
+     * @throws IdentityEventException   Identity event exception
+     */
+    private void storeMetadataToUserStore(UserStoreManager userStoreManager, String username, String tenantId,
+                                          String claimURI, String value, String eventName)
+            throws IdentityEventException {
+
+        String claimAttributeName = retrieveAttributeName(userStoreManager, tenantId, claimURI);
+        boolean executionResult;
+        try (Connection connection = getDBConnection(userStoreManager.getRealmConfiguration())) {
+            try {
+                if (isMetadataExistsInUserStore(connection, userStoreManager, tenantId, username, claimAttributeName)) {
+                    // update claim in user store
+                    executionResult = updateExistingMetadataInUserStore(connection, userStoreManager, tenantId,
+                            username, claimAttributeName, value);
+                } else {
+                    // add claim to user store
+                    executionResult = insertMetadataInUserStore(connection, userStoreManager, tenantId, username,
+                            claimAttributeName, value);
+                }
+            } catch (SQLException e) {
+                DatabaseUtil.rollBack(connection);
+                throw new IdentityEventException(String.format("Error while updating the user metadata in user store " +
+                        "related to %s event.", eventName), e);
+            }
+            connection.commit();
+        } catch (SQLException e) {
+            throw new IdentityEventException("Error while implementing database connection with user store.");
+        }
+
+        if (log.isDebugEnabled() && executionResult) {
+            log.debug(String.format("Successfully updated the user metadata related to %s event.", eventName));
+        }
+    }
+
+    /**
+     * Store user metadata in the identity store.
+     *
+     * @param username                  Username
```

Comment:
Make the method description consistent. There are places where the param descriptions are ended with a full-stop.

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/697/commits/dd3419fd69cc40e847e40ed9dfb29c867a3f0018